content/src/content/jcr_root/apps/core/wcm/sandbox/components/carousel/v1/carousel/clientlibs/site/js/carousel.js

```diff
@@ -0,0 +1,212 @@
+/*******************************************************************************
+ * Copyright 2018 Adobe Systems Incorporated
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ******************************************************************************/
+/* global Granite */
+(function() {
+    "use strict";
+
+    var NS = "cmp";
+    var IS = "carousel";
+
+    var selectors = {
+        self: "[data-" +  NS + '-is="' + IS + '"]'
+    };
+
+    function Carousel(config) {
+        var that = this;
+
+        if (config && config.element) {
+            init(config);
+        }
+
+        function init(config) {
+            // prevents multiple initialization
+            config.element.removeAttribute("data-" + NS + "-is");
+
+            cacheElements(config.element);
+            that._active = 0;
+
+            if (that._elements.item) {
+                refreshActive();
+                initControls();
+            }
+
+            if (Granite && Granite.author) {
+                new Granite.author.MessageChannel("cqauthor", window).subscribeRequestMessage("cmp.panelcontainer", function(message) {
```

Comment:
@jckautzmann, @vladbailescu - I would like to move this message subscription handling outside of the Carousel class and instead make the `navigate` method publicly accessible. But, we're missing the ability to use the API, how do we want access to our JS classes to work?

Some initial ideas:
`JQuery adaptTo: $('#myCarousel').adaptTo('cmp-carousel');`
`JQuery data: $('#myCarousel').data('cmp-carousel');`

Reply:
I think the `adaptTo` way would be expected for AEM.

Reply:
@vladbailescu - agree, I'm thinking we should hold off on adding this until we develop our widget component concept (component registration, property updates, event handling, API access etc.) and apply it to all UI components.